piet/src/text.rs

```diff
@@ -31,4 +32,46 @@ pub trait TextLayoutBuilder {
 pub trait TextLayout {
     /// Measure the advance width of the text.
     fn width(&self) -> f64;
+
+    /// Given a Point, determine the corresponding text position
+    fn hit_test_point(&self, point: Point) -> HitTestPoint;
+
+    /// Given a text position, determine the corresponding pixel location
+    /// (currently consider the text layout just one line)
+    fn hit_test_text_position(&self, text_position: usize, trailing: bool) -> Option<HitTestTextPosition>;
+}
+
+#[derive(Debug, Default, PartialEq)]
+/// return values for `hit_test_point`.
+/// - `metrics.text_position` will give you the text position.
+/// - `is_inside` indicates whether the hit test point landed within the text.
+/// - `is_trailing_hit` will always return `false` for now, until BIDI support is implemented.
+pub struct HitTestPoint {
+    pub metrics: HitTestMetrics,
+    pub is_inside: bool,
+
+    // removing until needed for BIDI or other.
+    //pub is_trailing_hit: bool,
 }
+
+/// return values for `hit_test_text_position`.
+/// - `point.x` will give you the x offset.
+#[derive(Debug, Default)]
+pub struct HitTestTextPosition {
+    pub point: Point,
+    pub metrics: HitTestMetrics,
+}
+
+#[derive(Debug, Default, PartialEq)]
+pub struct HitTestMetrics {
+    pub text_position: usize,
+    //length: u32,
```

Comment:
Probably should clean up these comments, possibly replace with something like "TODO: consider adding other metrics as needed, such as those provided in [DWRITE_HIT_TEST_METRICS](https://docs.microsoft.com/en-us/windows/win32/api/dwrite/ns-dwrite-dwrite_hit_test_metrics)".